datafusion/common/src/dfschema.rs

```diff
@@ -496,6 +497,15 @@ impl From<DFSchema> for SchemaRef {
     }
 }
 
+// Hashing refers to a subset of fields considered in PartialEq.
+#[allow(clippy::derive_hash_xor_eq)]
+impl Hash for DFSchema {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.fields.hash(state);
+        self.metadata.len().hash(state); // HashMap is not hashable
```

Comment:
I agree it is ok to just use the metadata's length to hash as it satisfies the EQ constraint

https://doc.rust-lang.org/std/hash/trait.Hash.html#hash-and-eq